pkg/agent/puller.go

```diff
@@ -57,6 +58,11 @@ func StartPuller(downloader Downloader, commands <-chan ModelOp, logger *zap.Sug
 		logger:      logger,
 	}
 	go puller.processCommands(commands)
+	for modelName, _ := range modelTracker {
```

Comment:
Unfortunately since both the Download and Load step happen within the Add command, waiting for the Add command to complete will not return after Downloads are complete, but after Loads are complete. I think the most obvious solution here would be instead wait until all models have directories in the `/mnt/models` directory, but that seems a little clunky. 

Could we possibly subdivide the Add command into two separate commands -> Download and Load? That way we can wait until a completion happens for all Downloads, while the Load commands run afterward.

Reply:
actually we want to wait for all the models to be loaded right? otherwise even we mark the agent to be ready the curl can still result in 404s since the models are not loaded yet.

Reply:
That's a fair point - I suppose the concern would be that it would take a long time for the model agent to be ready if there are a lot of models to load.